presto-main/src/main/java/com/facebook/presto/sql/analyzer/ExpressionAnalyzer.java

```diff
@@ -1121,6 +1110,15 @@ protected Type visitNode(Node node, StackableAstVisitorContext<Context> context)
             throw new SemanticException(NOT_SUPPORTED, node, "not yet implemented: " + node.getClass().getName());
         }
 
+        public Type visitGroupingOperation(GroupingOperation node, StackableAstVisitorContext<Context> context)
+        {
+            for (Expression columnArgument : node.getGroupingColumns()) {
+                process(columnArgument, context);
+            }
+            expressionTypes.put(node, INTEGER);
```

Comment:
INTEGER is 32 bit in Presto, so this should be BIGINT.

One interesting question is whether the type should depend on the number of columns. So, `grouping(c1, ..., c7) :: tinyint`, `grouping(c1, ..., c15) :: smallint`, `grouping(c1, ..., c31) :: integer`, `grouping(c1, ..., c63) :: bigint`, etc. What does the spec say about this?